### PR TITLE
Check for valid successor.

### PR DESCRIPTION
**Pull request by toregge**
@geirst : please review
@vekterli : FYI

### CODE REVIEW

File: searchlib/src/vespa/searchlib/attribute/dfa_fuzzy_matcher.h [hunk 1]
```diff
@@ -47,7 +49,7 @@ class DfaFuzzyMatcher {
                     return true;
                 }
                 _successor.resize(_prefix.size());
-                _successor.emplace_back(1);
+                _successor.emplace_back(beyond_unicode);
```
Comment (baldersheim) [inline searchlib/src/vespa/searchlib/attribute/dfa_fuzzy_matcher.h:52]: Belong in a hpp file due to size.
hardly needs iostream

**Review by baldersheim — APPROVED** [review]

**Review by geirst — APPROVED** [review]
👍

**Comment by geirst** [comment]
We already test the corner case where prefix length > target length (exact match) in _fuzzy_match_in_dictionary_with_prefix_size_ (https://github.com/vespa-engine/vespa/blob/master/searchlib/src/tests/attribute/dfa_fuzzy_matcher/dfa_fuzzy_matcher_test.cpp). With the assert added in _is_match()_ this test starts failing without the successor fix.
